enhancements/machine-config/manage-boot-images.md

```diff
@@ -319,6 +324,194 @@ When [MachineDeployments](https://cluster-api.sigs.k8s.io/developer/architecture
+  - **Marketplace-published**: Could be UPI or ROSA - apply normal skew enforcement rules, but require update to future images with proper metadata
+  - **RHCOS**: Follow regular skew rules (consider raising warning since it should be in the historical list)
+
+###### Future Bootimage Tracking (New Images)
+
+**Installer changes:**
+
+Create multiple JSON files of AMI variant metadata for different streams:
+- rhcos (standard IPI)
+- rosa
+- marketplace-ocp
+- marketplace-oke
+- marketplace-opp
+
+**Image creation:**
+
+- All variants are tagged with `variantType` metadata to enable deterministic stream detection
+- The tagging will be done with AWS's [tagSet](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Image.html) field, with `variantType as the Key and stream name as the value
+
+##### GCP
+
+###### History Bootimage Tracking (Existing Clusters)
+
+GCP boot images use format: `projects/<project-name>/global/images/<image-name>`, which we can use to parse the historical stream where possible.
+
+**GCP IPI installs:**
+- Installer images have project name `rhcos-cloud`
+
+**GCP Marketplace:**
+- Marketplace images have project name `redhat-marketplace-public`
+- The image name should contain the variant.
+
+**GCP UPI:**
+- UPI installations typically upload their own images
+- These fall into the non-managed case and require manual updates
+
+**Managed:**
+- No GCP managed service currently exists (see Open Questions regarding Dedicated)
+
+**Detection fallback:**
+
+If the image project parsing does not match any known streams or variants, we will default to requiring the user to do so manually.
+
+###### Future Bootimage Tracking (New Images)
+
+**Installer changes:**
+
+Create stream metadata files for:
+- rhcos (GCP IPI)
+- marketplace-ocp
+- marketplace-oke
+- marketplace-opp
+
+**Image creation:**
+
+While the image project can presently be used to detect stream, we should nevertheless do the same variantType tagging to make it explicit.
+
+For GCP, we will leverage [labels](https://cloud.google.com/compute/docs/labeling-resources) with `varianttype` as key and the stream name as value.
+
+###### Open Questions
+
+While GCP does not have a managed service offering, there is OpenShift Dedicated on GCP. We should check if they have special boot media in use.
+
+##### Azure
+
+###### History Bootimage Tracking (Existing Clusters)
+
+**Azure IPI installs and ARO (Azure Red Hat OpenShift)**
+
+ARO and regular IPI installs use the same images.
+
+Check the ProviderSpec `image` field structure:
+```yaml
+image:
+  offer: ''
+  publisher: ''
+  resourceID: /resourceGroups/...
+  sku: ''
+  version: ''
+```
+
+If `offer`, `publisher`, `sku`, and `version` are already set, this is already using the new unpaid marketplace image for IPI. We can check `publisher` for `azureopenshift`.
```

Comment:
This first sentence was slightly misleading when I read it at first. If `publisher` == `azureopenshift` then it is an unpaid marketplace image, `publisher` and `offer` will still be set for both paid and unpaid images. (All marketplace images need the complete URN).

Reply:
Will clarify this point. Basically what I tried to say is that `publisher` == `azureopenshift` is unpaid and the below `publisher` == `redhat` or `redhat-limited` is paid